fix: Re-align the bitstream

ikrivosheev opened the pull request:
I fixed the issue: https://github.com/Lonami/lzxd/issues/21

I read source code [gcab](https://github.com/GNOME/gcab/blob/master/libgcab/decomp.c) and [libmspack](https://github.com/kyz/libmspack/blob/master/libmspack/mspack/lzxd.c) and find an interesting thin. After Uncompressed block if length is not Odd then re-align the bitstream.

Also I added:

- [x] `#[derive(Copy, Clone, Debug, Eq)]` to `DecompressError` and `DecodeFailed`
- [x]  Fixed some clippy warnings

ikrivosheev committed:
fix: Re-align the bitstream

ikrivosheev commented:
@Lonami it will be cool if you release this. I need it for my job.

Lonami commented on src/bitstream.rs:
```diff
@@ -49,6 +49,17 @@ impl<'a> Bitstream<'a> {
         Ok(self.n & 1)
     }
 
+    pub fn read_byte(&mut self) -> Option<u8> {
+        if self.buffer.is_empty() {
+            return None;
+        }
+        let byte = self.buffer[0];
+        self.buffer = &self.buffer[1..];
+        self.remaining = 0;
+        self.n = 0;
+        Some(byte)
+    }
```
Shouldn't this behave more like `read_bits_oneword(8)`?

ikrivosheev replied:
@Lonami I think no. I tried to do this and it doesn't work.

Lonami commented on src/bitstream.rs:
```diff
@@ -147,13 +158,13 @@ impl<'a> Bitstream<'a> {
     }
 
-    pub fn align(&mut self) -> bool {
+    pub fn align(&mut self) -> Result<(), DecodeFailed> {
```
Maybe we should rename the method to make it clear that this operation will "trash" some data.

ikrivosheev replied:
Well, it doesn't look like trash, for me. And it was easier to debug, because I saw similar things in different code)

Lonami replied:
I mean trash in the sense that it reads 16 bits and drops them ("trashing" them, being unable to access them).

Reading `align`, it wouldn't be my first instinct that it does this. But maybe the name is fine.

Lonami reviewed (commented):
Nice investigation work and good clean-up. Ping me again once it's ready for review and then we can merge.

ikrivosheev commented:
@Lonami thank you for the review! Well, bug was in re-align buffer. I removed it from `Bitstream::read_raw`  and added it in `Lzxd::decompress_next` like in other implementations and it starts working!!

ikrivosheev commented:
Now it's ready for review and merge. It's work. I checked it on my computer.
